src/views/queryHistoryUI.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import VscodeWrapper from "../controllers/vscodeWrapper";
+import { IPrompter, IQuestion, QuestionTypes } from "../prompts/question";
+import { QueryHistoryProvider } from '../queryHistory/queryHistoryProvider';
+import { QueryHistoryNode } from '../queryHistory/queryHistoryNode';
+import * as LocalizedConstants from '../constants/localizedConstants';
+
+
+export enum QueryHistoryAction {
+    OpenQueryHistoryAction = 1,
+    RunQueryHistoryAction = 2
+};
+
+export interface QueryHistoryQuickPickItem extends vscode.QuickPickItem {
+    node: QueryHistoryNode;
+    action: any;
+};
+
+export class QueryHistoryUI {
+
+    constructor(
+        private _prompter: IPrompter,
+        private _vscodeWrapper: VscodeWrapper
+    ) {}
+
+    public convertToQuickPickItem(node: vscode.TreeItem): QueryHistoryQuickPickItem {
+        let historyNode = node as QueryHistoryNode;
+        let quickPickItem: QueryHistoryQuickPickItem = {
+            label: QueryHistoryProvider.limitStringSize(historyNode.queryString, true),
+            detail: `${historyNode.connectionLabel}, ${historyNode.timeStamp.toLocaleString()}`,
+            node: historyNode,
+            action: undefined,
+            picked: false
+        };
+        return quickPickItem;
+    }
+
+    private showQueryHistoryActions(node: QueryHistoryNode): Promise<string> {
+        let options = [{ label: LocalizedConstants.msgOpenQueryHistoryListing },
+            { label: LocalizedConstants.msgRunQueryHistoryListing }];
+        let question: IQuestion = {
+            type: QuestionTypes.expand,
+            name: 'question',
+            message: LocalizedConstants.msgChooseQueryHistoryAction,
+            choices: options
+        };
+        return this._prompter.promptSingle(question).then((answer: vscode.QuickPickItem) => {
+            if (answer) {
```

Comment:
Would be good to explicitly return undefined (and update the return type) if an answer isn't chosen.